messages/messages.py

```diff
@@ -0,0 +1,127 @@
+    csv_output_path = args.csv_output_path
+    
+    ICR_MESSAGES_COUNT = 200  # Number of messages to export in the ICR file
+    
+    # Convert date/time of messages to EAT
+    utc_key = "{} (Time) - {}".format(variable_name, flow_name)
+    eat_key = "{} (Time EAT) - {}".format(variable_name, flow_name)
+    inside_time_window = []
+    START_TIME = isoparse("2018-10-18T00+03:00")
+    END_TIME = isoparse("2018-10-27T00+03:00")
+
+    # Load data from JSON file
+    with open(json_input_path, "r") as f:
+        show_messages = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Filter out test messages sent by AVF.
+    show_messages = [td for td in show_messages if not td.get("test_run", False)]
+
+    # Filter for runs which contain a response to this week's question.
+    show_message_key = "{} (Text) - {}".format(variable_name, flow_name)
+    show_messages = [td for td in show_messages if show_message_key in td]
+
+    # Filter out messages sent outside the project run period
+    for td in show_messages:
+        utc_time = isoparse(td[utc_key])
+        eat_time = utc_time.astimezone(pytz.timezone("Africa/Nairobi")).isoformat()
+
+        td.append_data(
+            {eat_key: eat_time},
+            Metadata(user, Metadata.get_call_location(), time.time())
+        )
+
+        if START_TIME <= utc_time <= END_TIME:
+            inside_time_window.append(td)
+        else:
+            print("Dropping: {}".format(utc_time))
+
+    print("{}:{} Dropped as outside time/Total".format(len(show_messages) - len(inside_time_window), len(show_messages)))
+    show_messages = inside_time_window
+
+    # Output messages to a CSV file
+    IOUtils.ensure_dirs_exist_for_file(csv_output_path)
+    run_id_key = "{} (Run ID) - {}".format(variable_name, flow_name)
+    raw_text_key = "{} (Text) - {}".format(variable_name, flow_name)
+    with open(csv_output_path, "w") as f:
+        TracedDataCSVIO.export_traced_data_iterable_to_csv(show_messages, f, headers=["avf_phone_id", run_id_key, raw_text_key])
+
+    # Output messages to Coda
+    IOUtils.ensure_dirs_exist_for_file(coda_output_path)
+    run_id_key = "{} (Run ID) - {}".format(variable_name, flow_name)
+    raw_text_key = "{} (Text) - {}".format(variable_name, flow_name)
+    time_key = "{} (Time) - {}".format(variable_name, flow_name)
+    keys = [run_id_key, raw_text_key, time_key]
+    keymap = {run_id_key:"MessageID", raw_text_key:"Text", time_key:"CreationDateTimeUTC"}
+
+    messages_to_code = []
+    for td in show_messages:
+        td.append_data({"Labels":[]},  Metadata(user, Metadata.get_call_location(), time.time()))
+        messages_to_code.append({key:td[key] for key in keys})
+
+    for td in messages_to_code:
+        td_remapped = {}
+        for key in td:
+            td_remapped[keymap[key]] = td[key] 
+
+    with open(coda_output_path, "w") as f:
+        TracedDataCodaIO.export_traced_data_iterable_to_coda(show_messages, show_message_key, f)
```

Comment:
bit confused here. You re-mapped to use what look like Coda V2 keys, but then export using the V1 exporter...